src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -288,10 +291,65 @@ public static class Config {
         long keepAliveTimeoutSeconds;
         long maxReconnectDelayMs;
         long minReconnectDelayMs;
+
+        /**
+         * Map from bridge configuration to client configuration.
+         *
+         * @param bridgeConfig component configuration
+         * @return client configuration
+         */
+        public static Config fromBridgeConfig(BridgeConfig bridgeConfig) {
+            return Config.builder()
+                    .brokerUri(bridgeConfig.getBrokerUri())
+                    .optionsByTopic(bridgeConfig.getMqtt5RouteOptionsForSource(TopicMapping.TopicType.LocalMqtt))
+                    .clientId(bridgeConfig.getClientId())
+                    .sessionExpiryInterval(bridgeConfig.getSessionExpiryInterval())
+                    .maximumPacketSize(bridgeConfig.getMaximumPacketSize())
+                    .receiveMaximum(bridgeConfig.getReceiveMaximum())
+                    .ackTimeoutSeconds(bridgeConfig.getAckTimeoutSeconds())
+                    .connAckTimeoutMs(bridgeConfig.getConnAckTimeoutMs())
+                    .pingTimeoutMs(bridgeConfig.getPingTimeoutMs())
+                    .keepAliveTimeoutSeconds(bridgeConfig.getKeepAliveTimeoutSeconds())
+                    .maxReconnectDelayMs(bridgeConfig.getMaxReconnectDelayMs())
+                    .minReconnectDelayMs(bridgeConfig.getMinReconnectDelayMs())
+                    .build();
+        }
+
+        /**
+         * Determine if {@link LocalMqtt5Client} needs a restart, based on configuration changes.
+         *
+         * @param prevConfig previous config
+         * @param newConfig  new config
+         * @return true if the client needs a restart to apply config changes
+         */
+        public static boolean resetRequired(Config prevConfig, Config newConfig) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 327-337 into a separate method.